sqlglot/helper.py

```diff
@@ -435,3 +436,26 @@ def dict_depth(d: t.Dict) -> int:
 def first(it: t.Iterable[T]) -> T:
     """Returns the first element from an iterable (useful for sets)."""
     return next(i for i in it)
+
+
+def merge_ranges(ranges: t.List[t.Tuple[A, A]]) -> t.List[t.Tuple[A, A]]:
+    # type: ignore
+    if not ranges:
+        return []
+
+    # First, sort the ranges by starts
+    sorted_ranges = sorted(ranges, key=lambda x: x[0])  # type: ignore
```

Comment:
can't you just sort? by default it should sort on the first of the tuple

Reply:
chatgpt may have wrote this function...

let me clean it up.

Reply:
seriously? lol please don't introduce chatgpt stuff into this pristine hand crafted code base

Reply:
haha just that function.
I thought a thousand interviewees must've written a merge ranges function in their life.